src/components/generic/editable-field-components/MetaformNumberFieldComponent.tsx

```diff
@@ -0,0 +1,143 @@
+ */
+export class MetaformNumberFieldComponent extends React.Component<Props, State> {
+
+  /**
+   * Constructor
+   * 
+   * @param props component props
+   */
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+    };
+  }
+
+  /**
+   * Component render method
+   */
+  public render() {
+    const { classes, field, fieldId } = this.props;
+
+    const inputProps = {
+      min: field.min,
+      max: field.max,
+      step: 1
+    }
+
+    return (
+      <>
+        <FormControl>
+          <InputLabel
+            htmlFor={ fieldId }
+          >
+            { strings.editableFields.numberField }
+          </InputLabel>
+          <Input
+            type="number"
+            placeholder={ field.placeholder }
+            id={ fieldId }
+            name={ field.name }
+            title={ field.title }
+            required={ field.required }
+            readOnly={ field.readonly }
+            inputProps={ inputProps }
+            className={ classes.numberField }
+          />
+        </FormControl>
+        <FormControl>
+          <InputLabel
+            htmlFor={ fieldId }
+          >
+            { strings.editableFields.numberFieldMin }
+          </InputLabel>
+          <Input
+            type="number"
+            id={ fieldId }
+            className={ classes.numberField }
+            onChange={ this.handleNumberMinValueChange }
+          />
+        </FormControl>
+        <FormControl>
+          <InputLabel
+            htmlFor={ fieldId }
+          >
+            { strings.editableFields.numberFieldMax }
+          </InputLabel>
+          <Input
+            type="number"
+            id={ fieldId }
+            className={ classes.numberField }
+            onChange={ this.handleNumberMaxValueChange }
+          />
+        </FormControl>
+      </>
+    );
+  }
+
+  /**
+   * Event handler for minimum number value change
+   * 
+   * @param event new min number value
+  */ 
+  private handleNumberMinValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const changedMetaform = { ...this.props.metaform };
+    if(changedMetaform.sections){
+      changedMetaform.sections.forEach(section => {
+        section.fields?.forEach(field => {
+          if( field.type === "number"){
+            field.min = event.target.value as unknown as number;
+          }
+        })
+      });
+    }
+    
+    this.props.onMetaformUpdate(changedMetaform);
+  }
+
+  /**
+   * Event handler for maximum number value change
+   * 
+   * @param event new max number value
+  */ 
+  private handleNumberMaxValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const changedMetaform = { ...this.props.metaform };
+    if (changedMetaform.sections){
+      changedMetaform.sections.forEach(section => {
+        section.fields?.forEach(field => {
+          if( field.type === "number"){
+            field.max = event.target.value as unknown as number;
```

Comment:
Same here, why casting it twice?

Reply:
This was actually because of custom type field.max had, it had to be erased first and the cast to number. Better way could be Number(target.value), but since this whole array search is kind of taxing (and because Jari had an idea) the method is now removed and value change is made differently on outer component.